nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java

```diff
@@ -177,10 +177,11 @@ public List<WorkflowExecutor> getExecutors() {
         });
   }
 
-  public void markShutdown() {
+  public void markShutdown(boolean graceful) {
     try {
-      jdbc.update("update nflow_executor set expires=current_timestamp, stopped=current_timestamp, recovered=null where "
-          + getExecutorGroupCondition() + " and id = ?", getExecutorId());
+      var sql = graceful ? "set expires=current_timestamp, stopped=current_timestamp, recovered=current_timestamp" :
+              "set expires=" + sqlVariants.currentTimePlusSeconds(timeoutSeconds) + ", stopped=current_timestamp, recovered=null";
+      jdbc.update("update nflow_executor " + sql + " where id = " + getExecutorId());
```

Comment:
More reabable and uses prepared statement:
```suggestion
      var expires = graceful ? "current_timestamp" : sqlVariants.currentTimePlusSeconds(timeoutSeconds);
      jdbc.update("update nflow_executor set expires=?, stopped=current_timestamp, recovered=current_timestamp where id = ?",
          expires, getExecutorId());
```

Reply:
I don't think `sqlVariants.currentTimePlusSeconds` works as query parameter. All other uses are inlined.

Reply:
True. But the prepared statement argument still holds.